main/src/sphexa/propagator.hpp

```diff
@@ -172,132 +179,160 @@ class HydroProp final : public Propagator<DomainType, ParticleDataType>
         findNeighborsSfc<T, KeyType>(
             first, last, ngmax_, d.x, d.y, d.z, d.h, d.codes, d.neighbors, d.neighborsCount, domain.box());
         timer.step("FindNeighbors");
+#ifdef USE_CUDA
+        CHECK_CUDA_ERR(cudaMemcpy(d.devPtrs.d_codes, d.codes.data(), size_np_CodeType, cudaMemcpyHostToDevice));
+        timer.step("  * GPU CudaCopyIn Sync HostToDevice: codes");
+#endif
+
+#ifdef USE_CUDA
+        CHECK_CUDA_ERR(cudaMemcpy(d.devPtrs.d_x, d.x.data(), size_np_T, cudaMemcpyHostToDevice));
```

Comment:
As a suggestion to clean this a bit out in the future: these copies could be added in a single function (f.e. Host2Device) that takes the different fields as inputs. Then we can have a single line instruction while still keeping explicitly written the data that is being transferred.